src/components/GoalCriteria/GoalCriteria.tsx

```diff
@@ -416,6 +425,37 @@ export const GoalCriteria: React.FC<GoalCriteriaProps> = ({
         );
     }, [sortedCriteriaItems]);
 
+    const mapCriteriaToValues = useCallback((criteria: CriteriaItemValue): CriteriaFormData => {
+        if (criteria.criteriaGoal) {
+            setEditMode('goal');
+
+            return {
+                mode: 'goal',
+                id: criteria.id,
+                title: criteria.title,
+                selected: {
+                    id: criteria.criteriaGoal.id,
+                    title: criteria.criteriaGoal.title,
+                    stateColor: criteria.criteriaGoal.stateColor,
+                },
+                weight: criteria.weight > 0 ? String(criteria.weight) : '',
+            };
+        }
+
+        setEditMode('simple');
+
+        return {
+            mode: 'simple',
+            id: criteria.id,
+            title: criteria.title,
+            weight: criteria.weight > 0 ? String(criteria.weight) : '',
+        };
+    }, []);
+
+    const handleChangeInput = useCallback((val = '') => {
```

Comment:
val: string

Reply:
Nope, cuz it can be undefine, also TS inferred type from default variable value